### ISSUE TITLE
Remove `PreferToOverPairSyntax` for 2.0

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#7920]
`PreferToOverPairSyntax` rule is controversial. There are cases where `Pair()` is better than `to` and it's difficult to add exclusions to those cases. And we even have this rule enabled by default.

An example can be found at #7916

But here is another:

```kotlin
val foo = "hello hello hello hello hello ... hello hello hello hello" to
  "world"

// vs

val foo = Pair(
  "hello hello hello hello hello ... hello hello hello hello",
  "world",
)
```

I vote to remove this rule. I agree that in general `to` is better than `Pair()` but a rule that raises a lot of false positives is not worth to maintain, I think.

**Comment by atulgpt** [comment]
Instead, we can disable the rule by default and handle the cases as they come like other rules. This way people who still want to use the rule can enable this rule.

**Comment by BraisGabin** [comment]
The problem with that is that we need to maintain the rule. The rule is there and if someone thinks it's good they can just copy&paste it and move it to a third party rule set.

Maintaining a rule is work. Even if the PRs are done by contributors decide if a change is worth or not on an issue is time consuming. And the same to review the PR. I think that we should focus our efforts on other parts than a rule that doesn't seem to have a good way to be "correct".
